STMReco/src/MWDAlg.cc

```diff
@@ -0,0 +1,214 @@
+#include "Offline/STMReco/inc/MWDAlg.hh"
+#include <chrono>
+#include <thread>
+
+namespace mu2e {
+
+  MWDAlg::MWDAlg(){
+    M = 1000;
+    L = 500;
+    tau = 50000;  // in ns
+    nsigma_cut = 9;
+    thresholdgrad = -0.3;
+    fADC = 320.0520833313; // in MHz
+    cut_mode = 1;
+    fixed_cut_parameter = -1000.0;
+  }
+
+  MWDAlg::MWDAlg(double _M, double _L, double _tau, double _nsigma_cut, double _thresholdgrad, double _fADC, int _cut_mode, double _fixed_cut_parameter){
+    M = _M;
+    L = _L;
+    tau = _tau;
+    nsigma_cut = _nsigma_cut;
+    thresholdgrad = _thresholdgrad;
+    fADC = _fADC;
+    cut_mode = _cut_mode;
+    fixed_cut_parameter = _fixed_cut_parameter;
+  }
+
+  std::string MWDAlg::print() {
+    std::stringstream ss;
+    ss << "M             = " << M << "\n"
+       << "L             = " << L << "\n"
+       << "tau           = " << tau << "\n"
+       << "nsigma_cut    = " << nsigma_cut << "\n"
+       << "fixed_cut     = " << fixed_cut_parameter << "\n"
+       << "cut_mode      = " << cut_mode << "\n"
+       << "nsigma_cut    = " << nsigma_cut << "\n"
+       << "thresholdgrad = " << thresholdgrad << "\n"
+       << "fADC [MHz]    = " << fADC << "\n" ;
+    return ss.str();
+  }
+
+  peaks* MWDAlg::find_peaks(double baseline_mean, double baseline_rms, double time_offset) {
+
+    peaks* peak_data = new peaks();
+    peak_data->npeaks = 0;
+
+    if (cut_mode == 1)
+      threshold_cut = baseline_mean - baseline_rms*nsigma_cut;
+    else if (cut_mode == 2)
+      threshold_cut = fixed_cut_parameter;
+    else {
+      std::cout << "ERROR:: find_peaks() : unknown cut mode .... " << cut_mode << std::endl;
+      exit(-1);
+    }
+
+    //    if (cut_mode == 1)
```

Comment:
If these printouts are useful I suggest to add a 'diaglevel' member and enable them when set.